### ISSUE TITLE
[BUG] Set-PnPListItem -ContentType is not updating the content type

### ISSUE THREAD

**Issue by rtonerii** [pnp/powershell#4161]
## **Bug Report**

**Title:** Set-PnPListItem -ContentType Not Updating Content Type - Critical Impact on Mass Retagging in Public SharePoint Online

**Summary:**
This issue is severely impacting a mass retagging project involving millions of records in a Public SharePoint Online environment. The `Set-PnPListItem` cmdlet fails to update content types as expected, causing significant delays and disruptions.

**Environment:**
* PnP PowerShell Version: 2.5.0
* PowerShell Version: 7.4.4
* Operating System: Windows 10 and Windows 11
* VS Code: 1.92.1
* PowerShell Module: v2024.2.2
* SharePoint Environment: Public SharePoint Online

**Expected Behavior:**
The `Set-PnPListItem` cmdlet should accurately update the content type of a list item when the `-ContentType` parameter is specified in a Public SharePoint Online environment.

**Actual Behavior:**
The content type of the list item remains unchanged despite using the `-ContentType` parameter in a Public SharePoint Online environment.

**Steps to Reproduce:**
```powershell
$listDefaultContentType = Get-PnPContentType | Where-Object {$_.Name -eq "Specific Type"}
Set-PnPListItem -List "SomeList" -Identity 11 -Values @{"Text1"="SomeText"} -UpdateType SystemUpdate -ContentType $listDefaultContentType
```

**Tried Alternatives:**
1. Using `ContentTypeId` property:
   ```powershell
   Set-PnPListItem -List "SomeList" -Identity 11 -Values @{"Text1"="SomeText","ContentTypeId"= "SomeContypeId"} -UpdateType SystemUpdate
   ```
2. Omitting `-Values` parameter:
   ```powershell
   Set-PnPListItem -List "SomeList" -Identity 11 -ContentType $listDefaultContentType
   ```
3. Using `-UpdateType` without `-Values`:
   ```powershell
   Set-PnPListItem -List "SomeList" -Identity 11 -UpdateType SystemUpdate -ContentType $listDefaultContentType
   ```

**Error/Output:**
No errors are generated, but the content type remains unchanged. The command output is standard:
```
Id  Title                                    GUID
--  -----                                    ----
11  Item Title                               "Item GUID"
```

**Impact:**
This issue is severely impacting a large-scale retagging project, affecting millions of records and causing significant delays in a Public SharePoint Online environment.

**Additional Information:**
* The issue has been reproduced in both VS Code and PowerShell console environments.
* The `-UpdateType SystemUpdate` is required for audit purposes.

**Comment by jackpoz** [comment]
I cannot reproduce this issue. I created 3 list items with 3 different content types:
![image](https://github.com/user-attachments/assets/8d2ad9b2-f396-47bb-8b93-d90662c88b75)

Then tried to change the content type to "Event" for 2 of them and it worked:
![image](https://github.com/user-attachments/assets/7e868833-16ad-477d-a7db-6b257475ac04)

Tried with and without SystemUpdate:
![image](https://github.com/user-attachments/assets/ba29162f-4dfb-4829-aac5-64a9a523f138)

Are you sure that the content type is available in the list ? I would suggest to you `Get-PnPContentType -List "SomeList"`  and verify that the content type has been added to the list.

**Comment by rtonerii** [comment]
I can see the content type; I got permission to try and manual update a content type on a record.  Once I tested it the content type was available for selecting on the list item and it saved, one second later it went back to the old content type.

It looks like there is something going on with the client's content, they are opening a ticket to see if this can be identified.

We can close this bug as it seems to be something in the environment.
